make warn_deprecated_hint_without_locality AsError (#16004)
coq/coq: doc/sphinx/addendum/type-classes.rst

   .. deprecated:: 8.14

>       The default value for instance locality will change in a future release.

Reviewer: Is it still a future change?  If not, drop the sentence.

Author: Yes, we only make the deprecation more discouraging in this PR. The default value will change next iteration.

>       Instances added outside of sections without an explicit locality are now
>       deprecated. We recommend using :attr:`export` where possible. This warning
>       will be treated as an error by default.

Reviewer: ```suggestion
      Instances added outside of sections without an explicit locality are
      deprecated. We recommend using :attr:`export` where possible. This warning
      is treated as an error by default.
```

      The default value for instance locality will change in a future
      release. Instances added outside of sections without an explicit
      locality are now deprecated. We recommend using :attr:`export`

>       where possible. This warning will error by default.

Reviewer: ```suggestion
      where possible. This warning will be treated as an error by default.
```

>       where possible. This warning will raise an error by default.

Reviewer: "This warning will raise an error by default." seems poorly worded--is it a warning or an error?  Perhaps you mean that the "warning" can be disabled through the standard mechanism and you can in include hyperlink to the command for doing so (which I don't remember offhand).  Perhaps you should refer to it as a "message" rather than a "warning"?

Author: Apparently warnings can be turned into errors as well as be disabled. This means that a user would be able to change its default behaviour back to be an error as well as disabling/enabling it.

Author: I'll change the wording to "This warning will error by default.".

Reviewer: Not to be too picky, but "error" is not a verb in the dictionaries I looked in.
